dbos/_dbos.py

```diff
@@ -945,16 +945,45 @@ def fn() -> None:
         return cls.retrieve_workflow(workflow_id)
 
     @classmethod
-    def restart_workflow(cls, workflow_id: str) -> WorkflowHandle[Any]:
+    def get_max_function_id(cls, workflow_uuid: str) -> Optional[int]:
+        max_transactions = _get_dbos_instance()._app_db.get_max_function_id(
+            workflow_uuid
+        )
+        max_operations = _get_dbos_instance()._sys_db.get_max_function_id(workflow_uuid)
+
+        if max_transactions is None and max_operations is None:
+            return None
+        if max_transactions is None:
+            return max_operations
+        if max_operations is None:
+            return max_transactions
+        return max(max_transactions, max_operations)
+
+    @classmethod
+    def restart_workflow(
```

Comment:
Again,w e should add a new `DBOS.fork_workflow`, mark `restart_workflow` as deprecated, and make it call fork(1).